gdk/common/recipe_file/CaseInsensitiveRecipe.py

```diff
@@ -0,0 +1,54 @@
+from requests.structures import CaseInsensitiveDict
+from gdk.common.recipe_file.Recipe import Recipe
+
+
+class CaseInsensitiveRecipe:
+    def __convert_nested_dict(self, recipe: CaseInsensitiveDict) -> CaseInsensitiveDict:
```

Comment:
Nit: don't call this a recipe given it could be whatever
```suggestion
    def __convert_nested_dict(self, recipe: CaseInsensitiveDict) -> CaseInsensitiveDict:
```